engines/director/lingo/lingo-builtins.cpp

```diff
@@ -1737,9 +1737,18 @@ void LB::b_editableText(int nargs) {
 }
 
 void LB::b_erase(int nargs) {
-	g_lingo->printSTUBWithArglist("b_erase", nargs);
+	Datum d = g_lingo->pop();
+	CastMember *eraseCast = g_director->getCurrentMovie()->getCastMember(d.asMemberID());
 
-	g_lingo->dropStack(nargs);
+	if (eraseCast) {
+		eraseCast->_erase = true;
+
+		for (uint i = 0; i < g_director->getCurrentMovie()->getScore()->_channels.size(); i++) {
```

Comment:
maybe the `g_director->getCurrentMovie()->getScore()` or even `g_director->getCurrentMovie()->getScore()->_channels` should be in a local variable as pointer or reference to reduce code duplication here.